test/integration/testPutGet.js

```diff
@@ -857,3 +861,315 @@ describe('PUT GET test multiple files', function () {
+        size: 0
+      },
+      {
+        name: 'large size file',
+        putQuery: `PUT ${largeFilePath} ${stage} AUTO_COMPRESS=FALSE`,
+        size: 100*1024*1024
+      }
+    ];
+
+  const createItCallback = function (testCase) {
+    return function (done) {
+      {
+        async.series(
+          [
+            function (callback) {
+              connection.execute({
+                sqlText: testCase.putQuery,
+                complete: function (err, stmt) {
+                  if(err){
+                    callback(err);
+                  }else {
+                    const stream = stmt.streamRows();
+                    stream.on('error', function (err) {
+                      callback(err);
+                    });
+                    stream.on('data', function (row) {
+                      // Check the file is correctly uploaded
+                      assert.strictEqual(row['status'], UPLOADED);
+                      assert.strictEqual(row.targetSize, testCase.size);
+                    });
+                    stream.on('end', function () {
+                      callback();
+                    });
+                  }
+                }
+              });
+            },
+            function (callback) {
+              connection.execute({
+                sqlText: getQuery,
+                complete: function (err, stmt) {
+                  if(err){
+                    callback(err);
+                  }else {
+                    const stream = stmt.streamRows();
+                    stream.on('error', function (err) {
+                      callback(err);
+                    });
+                    stream.on('data', function (row) {
+                      assert.strictEqual(row['status'], DOWNLOADED);
+                      // Verify the size of the file
+                      const file = path.join(tmpDir, row.file);
+                      const size = fs.statSync(file).size;
+                      assert.strictEqual(size, testCase.size);
+                    });
+                    stream.on('end', function () {
+                      callback();
+                    });
+                  }
+                }
+              });
+            }
+          ],
+          done
+        );
+      }
+    };
+  };
+
+  for (let index = 0; index < testCases.length; index++) {
+    const testCase = testCases[index];
+    it(testCase.name, createItCallback(testCase));
+  }
+});
+
+describe('PUT GET test with error', function () {
+  let connection;
+  const stage = `@${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME}`;
+  const stage_not_exist = `@${DATABASE_NAME}.${SCHEMA_NAME}.%NONEXISTTABLE`;
+  const createTable = `create or replace table ${TEMP_TABLE_NAME} (${COL1} STRING, ${COL2} STRING, ${COL3} STRING)`;
+  const removeFile = `REMOVE ${stage}`;
+  const dropTable = `DROP TABLE IF EXISTS ${TEMP_TABLE_NAME}`;
+
+  // Create a temp file without specified file extension
+  const tmpFile = tmp.fileSync(); 
```

Comment:
let's create file in before

Reply:
I think these ones we could keep it in describe. Otherwise we have to use let instead of const.

Reply:
Still we should not perform any actions in describe

Reply:
The same situation like in line 968.

Reply:
updated.